src/cfnlint/core.py

```diff
@@ -240,6 +241,18 @@ def run_checks(filename, template, rules, regions, mandatory_rules=None, validat
             )
             raise InvalidRegistryTypesException(msg, 32)
 
+    # MODULE is the only accepted value right now, but eventually other values will be supported such as RESOURCE
+    for registry_type in validate_registry_types:
+        if registry_type == 'MODULE':
+            # Check the presence of modules in the template
+            template_obj = Template(filename, template, regions)
+            modules = template_obj.get_modules()
+            if modules:
+                schema_manager = SchemaManager(filename, template, regions)
+                # For each module extracted from the template, verify if it's already locally cached
+                for module in modules:
```

Comment:
(not necessary, but future idea depending on speed)
could probably be done in [parallel](https://docs.python.org/3/library/multiprocessing.html) like we did for https://github.com/aws-cloudformation/cfn-lint/issues/1249#issuecomment-562801400